src/index.js

```diff
@@ -0,0 +1,243 @@
+const inquirer = require("inquirer");
+const {
+  viewDepartments,
+  viewRoles,
+  viewEmployees,
+  addDepartment,
+  addRole,
+  addEmployee,
+  updateEmployee,
+  deleteRole,
+  deleteDepartment,
+  deleteEmployee,
+  viewBudget,
+  viewByDepartment,
+  viewByManager,
+} = require("./db/queries");
+const { initialQuestion, deptQuestion } = require("./questions");
+const Db = require("./middleware/db");
+const {
+  generateDepartmentChoices,
+  generateRoleChoices,
+  generateManagerChoices,
+  generateEmployeeChoices,
+} = require("./utils");
+
+const title = `
+                                                                    
+▄███▄   █▀▄▀█ █ ▄▄  █    ████▄ ▀▄    ▄ █▀▄▀█ ▄███▄      ▄     ▄▄▄▄▀ 
+█▀   ▀  █ █ █ █   █ █    █   █   █  █  █ █ █ █▀   ▀      █ ▀▀▀ █    
+██▄▄    █ ▄ █ █▀▀▀  █    █   █    ▀█   █ ▄ █ ██▄▄    ██   █    █    
+█▄   ▄▀ █   █ █     ███▄ ▀████    █    █   █ █▄   ▄▀ █ █  █   █     
+▀███▀      █   █        ▀       ▄▀        █  ▀███▀   █  █ █  ▀      
+          ▀     ▀                        ▀           █   ██         
+                                                                    
+               ▄▄▄▄▀ █▄▄▄▄ ██   ▄█▄    █  █▀ ▄███▄   █▄▄▄▄          
+            ▀▀▀ █    █  ▄▀ █ █  █▀ ▀▄  █▄█   █▀   ▀  █  ▄▀          
+                █    █▀▀▌  █▄▄█ █   ▀  █▀▄   ██▄▄    █▀▀▌           
+               █     █  █  █  █ █▄  ▄▀ █  █  █▄   ▄▀ █  █           
+              ▀        █      █ ▀███▀    █   ▀███▀     █            
+                      ▀      █          ▀             ▀        `;
+
+const db = new Db({
+  host: process.env.DB_HOST || "localhost",
+  user: process.env.DB_USER || "root",
+  password: process.env.DB_PASSWORD || "password123",
+  database: process.env.DB_NAME || "tracker_db",
+});
+
+const start = async () => {
+  console.log(title);
+
+  await db.start();
+
+  let inProgress = true;
+
+  while (inProgress) {
+    const departments = await db.query("SELECT * FROM tracker_db.department");
+    const roles = await db.query("SELECT * FROM tracker_db.role");
+    const managers = await db.query(
+      "SELECT DISTINCT A.id, A.first_name, A.last_name FROM tracker_db.employee A, tracker_db.employee B WHERE A.id = B.manager_id"
+    );
+    const employees = await db.query("SELECT * FROM tracker_db.employee");
+
+    let answers = await inquirer.prompt(initialQuestion);
+
+    if (answers.initial === "depts") {
+      await viewDepartments(db);
+    }
+    if (answers.initial === "roles") {
+      await viewRoles(db);
+    }
+    if (answers.initial === "emps") {
+      await viewEmployees(db);
+    }
+    if (answers.initial === "addDept") {
+      const answer = await inquirer.prompt(deptQuestion);
+      const value = answer.name;
+      await addDepartment(db, value);
+    }
+    if (answers.initial === "addRole") {
+      const roleQuestions = [
+        {
```

Comment:
Add verification onto the questions to ensure that they cannot be left blank  as this throws a SQL error